include/abb_librws/rws_rapid.h

```diff
@@ -342,33 +346,188 @@ struct RAPIDRecord : public RAPIDSymbolDataAbstract
+
+    ss << "[";
+
+    for (size_t i = 0; i < container_.size(); ++i)
+    {
+      ss << container_.at(i).constructString();
+
+      if (i != container_.size() - 1)
+      {
+        ss << ",";
+      }
+    }
+
+    ss << "]";
+
+    return ss.str();
+  }
+
+  /**
+   * \brief A method for parsing a RAPID symbol data value string.
    *
-   * \param input for the string to search.
-   * \param character specifying the character to search for.
+   * \param value_string containing the string to parse.
+   * 
+   * \throws invalid_argument when value_string represent an array of a different size
+   */
+  void parseString(const std::string& value_string) override
+  {
+    std::vector<std::string> substrings = extractDelimitedSubstrings(value_string, '[', ']', ',');
+
+    if(array_size != substrings.size())
+    {
+      std::stringstream ss;
+      ss << "Cannot parse: The passed value_string=\"";
+      ss << value_string;
+      ss << "\" represents an array with size=";
+      ss << substrings.size();
+      ss << " different from the array size=";
+      ss << array_size;
+    
+      std::string exception_message = ss.str();
+
+      throw std::invalid_argument {exception_message};
+    }
+
+    for (size_t i = 0; i < substrings.size(); ++i)
+    {
+      container_.at(i).parseString(substrings.at(i));
+    }
+  }
+
+  /**
+   * \brief A method for getting the type of item stored in the RAPID array.
    *
-   * \return unsigned int containing the number of times the character occurs.
+   * \return std::string containing the type.
    */
-  unsigned int countCharInString(std::string input, const char character);
+  std::string getType() const override
+  {
+    return value_type().getType();
+  }
 
   /**
-   * \brief A method to extract delimited substrings in a string.
+   * \brief Operator for copying the RAPID array to another RAPID array.
    *
-   * \param input containing the string with delimited substrings.
+   * \param other containing the RAPID array to copy.
    *
-   * \return std::vector<std::string> containing the extracted substrings.
+   * \return RAPIDArray& containing the copy.
    */
-  std::vector<std::string> extractDelimitedSubstrings(const std::string& input);
+  RAPIDArray<value_type, array_size>& operator=(const RAPIDArray<value_type, array_size>& other)
+  {
+    if (this != &other)
+    {
+      for (size_t i = 0; i < array_size; ++i)
+      {
+        container_ = other.container_;
+      }
+    }
+
+    return *this;
+  }
 
   /**
-   * \brief The record's type name.
+   * \brief Mutably access an element of the underlying using its index
+   * 
+   * \param index The index of the item to access in the underlying container
+   * \return value_type& The value at the given index
+   * 
+   * \throws invalid_argument when the index is larger or equal than the array size
    */
-  std::string record_type_name_;
+  value_type& operator[](const size_t index) {
+    ensureIndexIsValid(index);
```

Comment:
`operator[]` is not supposed to check array bounds, this is why you have `at()`.

Reply:
I did not know this convention. Thanks for the information